include/Basic/SourceManager.hpp

```diff
@@ -0,0 +1,236 @@
+#ifndef GLU_SOURCE_MANAGER_HPP
+#define GLU_SOURCE_MANAGER_HPP
+
+#include <llvm/ADT/SmallVector.h>
+#include <llvm/ADT/StringRef.h>
+#include <llvm/Support/MemoryBuffer.h>
+#include <llvm/Support/VirtualFileSystem.h>
+
+#include "SourceLocation.hpp"
+
+namespace glu {
+
+///
+/// @class ContentCache
+/// @brief A file loaded will be cached in this class.
+///
+/// The ContentCache class is responsible for caching the content of a file that
+/// has been loaded. It is used by the SourceManager to store the content of a
+/// file. The content of a file is stored in a MemoryBuffer object which is
+/// owned by the ContentCache.
+///
+class alignas(8) ContentCache {
+
+private:
+    llvm::StringRef _fileName;
```

Comment:
this will be a reference to a temporary, which will be deallocated. it will crash. use std::string to keep a copy